pkg/gather/tasks_processing_test.go

```diff
@@ -0,0 +1,202 @@
+package gather
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/openshift/insights-operator/pkg/gatherers"
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+func Test_HandleTasksConcurrently(t *testing.T) {
+	const N = 3
+
+	var tasks []Task
+
+	for i := 0; i < N; i++ {
+		i := i
+		tasks = append(tasks, Task{
+			Name: fmt.Sprintf("task_%v", i),
+			F: gatherers.GatheringClosure{
+				Run: func(context.Context) ([]record.Record, []error) {
+					var records []record.Record
+
+					for j := 0; j < 2; j++ {
+						records = append(records, record.Record{
+							Name: fmt.Sprintf("task_%v_record_%v", i, j),
+						})
+					}
+
+					return records, nil
+				},
+				CanFail: false,
+			},
+		})
+	}
+
+	results := handleTasksConcurrentlyGatherTasks(tasks)
+	for i, _ := range results {
+		results[i].TimeElapsed = 0
+	}
+
+	assert.Len(t, results, 3)
+	assert.ElementsMatch(t, results, []GatheringFunctionResult{
+		{
+			FunctionName: "task_0",
+			Records: []record.Record{
+				{Name: "task_0_record_0"},
+				{Name: "task_0_record_1"},
+			},
+		},
+		{
+			FunctionName: "task_1",
+			Records: []record.Record{
+				{Name: "task_1_record_0"},
+				{Name: "task_1_record_1"},
+			},
+		},
+		{
+			FunctionName: "task_2",
+			Records: []record.Record{
+				{Name: "task_2_record_0"},
+				{Name: "task_2_record_1"},
+			},
+		},
+	})
+}
+
+// TODO: consider removing. The idea of the test is to check that execution is not sequential,
+// but the problem is that it's not guaranteed that it will take some specific time.
+// Maybe it should be an integration test
```

Comment:
+1 for removing and creating an integration test

Reply:
Replaced with a todo :)